THNN.lua

```diff
@@ -73,6 +73,13 @@ local replacements =
     }
 }
 
+-- gsub(s, 'real', 'float') changes accreal to accfloat.
+-- typedef accfloat ahead of time.
+ffi.cdef("typedef double accfloat;")
```

Comment:
But it is `float` in the CUDA version for performance reasons. I talked to soumith about it and decided to not change the way they are right now.